### PR TITLE
Async UART support

### PR DESCRIPTION
**Pull request by ivmarkov**

### CODE REVIEW

File: src/can.rs [hunk 1]
```diff
@@ -615,6 +615,14 @@ where
         })
     }
 
+    pub fn driver(&self) -> &CanDriver<'d> {
```
Comment (MabezDev) [inline src/can.rs:618]: Looks like these changes snuck in?

File: src/uart.rs [hunk 2]
```diff
@@ -752,30 +766,53 @@ impl<'d> UartDriver<'d> {
...
+                queue: rx_queue,
                 _p: PhantomData,
             },
         )
     }
 
     /// Read multiple bytes into a slice
-    pub fn read(&self, buf: &mut [u8], delay: TickType_t) -> Result<usize, EspError> {
-        self.rx().read(buf, delay)
+    pub fn read(&self, buf: &mut [u8], timeout: TickType_t) -> Result<usize, EspError> {
```
Comment (MabezDev) [inline src/uart.rs:796]:
```suggestion
    pub fn read(&self, buf: &mut [u8], timeout: T) -> Result<usize, EspError> 
    where T: Into<TickType_t>
    {
```

What do you think about this?

File: src/can.rs (hunk 1 above)
Comment (ivmarkov) [reply src/can.rs:618]: Yeah. Being lazy over there. We need these changes, but I should've opened a separate PR for these.

File: src/uart.rs (hunk 2 above)
Comment (ivmarkov) [reply src/uart.rs:796]: Sure - that would improve the ergonomics of the API. It is just that we then need to apply it everywhere where `TockType_t` is currently used. Perhaps a separate PR?

**Pull request merged**